tests/glsl-intrinsic/shader-subgroup/shader-subgroup-arithmetic.slang

```diff
@@ -0,0 +1,200 @@
+        //&& testVLogical<int8_t, 3>()
+        //&& testVLogical<int8_t, 4>()
+        //&& test1Logical<int16_t>()
+        //&& testVLogical<int16_t, 2>()
+        //&& testVLogical<int16_t, 3>()
+        //&& testVLogical<int16_t, 4>()
+        //&& test1Logical<int64_t>()
+        //&& testVLogical<int64_t, 2>()
+        //&& testVLogical<int64_t, 3>()
+        //&& testVLogical<int64_t, 4>()
+        && test1Logical<uint>()
+        && testVLogical<uint, 2>()
+        && testVLogical<uint, 3>()
+        && testVLogical<uint, 4>()
+        && test1Logical<bool>()
+        && testVLogical<bool, 2>()
+        && testVLogical<bool, 3>()
+        && testVLogical<bool, 4>()
+        ;
+}
+
+__generic<T : __BuiltinArithmeticType>
+bool test1Arithmetic() {
+    return true
+        && subgroupAdd(T(1)) == T(local_size_x_v) // 32
+           && subgroupMul(T(1)) == T(1)
+        && subgroupMin(T(1)) == T(1)
+        && subgroupMin(T(1)) == T(1)
+        && subgroupExclusiveAdd(T(1)) == T(3)
+        && subgroupExclusiveMul(T(1)) == T(1)
+        && subgroupExclusiveMin(T(1)) == T(1)
+        && subgroupExclusiveMax(T(1)) == T(1)
+#ifdef TEST_when_glsl_subgroupInclusiveAdd_is_not_bugged
+           && subgroupInclusiveAdd(T(1)) == T(4)
+#endif // #ifdef TEST_when_glsl_subgroupInclusiveAdd_is_not_buggeds
+        && subgroupInclusiveMul(T(1)) == T(1)
+        && subgroupInclusiveMin(T(1)) == T(1)
+        && subgroupInclusiveMax(T(1)) == T(1)
+        ;
+}
+__generic<T : __BuiltinArithmeticType, let N : int>
+bool testVArithmetic() {
+    typealias gvec = vector<T, N>;
+
+    return true
+        && subgroupAdd(gvec(T(1))) == gvec(T(local_size_x_v)) // 32
+        && subgroupMul(gvec(T(1))) == gvec(T(1))
+        && subgroupMin(gvec(T(1))) == gvec(T(1))
+        && subgroupMin(gvec(T(1))) == gvec(T(1))
+        && subgroupExclusiveAdd(gvec(T(1))) == gvec(T(3))
+        && subgroupExclusiveMul(gvec(T(1))) == gvec(T(1))
+        && subgroupExclusiveMin(gvec(T(1))) == gvec(T(1))
+        && subgroupExclusiveMax(gvec(T(1))) == gvec(T(1))
+#ifdef TEST_when_glsl_subgroupInclusiveAdd_is_not_bugged
+        && subgroupInclusiveAdd(gvec(T(1))) == gvec(T(4))
+#endif // #ifdef TEST_when_glsl_subgroupInclusiveAdd_is_not_bugged 
+        && subgroupInclusiveMul(gvec(T(1))) == gvec(T(1))
+        && subgroupInclusiveMin(gvec(T(1))) == gvec(T(1))
+        && subgroupInclusiveMax(gvec(T(1))) == gvec(T(1))
+        ;
+}
+
+bool testArithmetic() {
+    return true
+        && test1Arithmetic<float>()
+        && testVArithmetic<float, 2>()
+        && testVArithmetic<float, 3>()
+        && testVArithmetic<float, 4>()
+        //&& test1Arithmetic<double>() // hang
+        //&& testVArithmetic<double, 2>()
+        //&& testVArithmetic<double, 3>()
+        //&& testVArithmetic<double, 4>()
+        //&& test1Arithmetic<half>()
+        //&& testVArithmetic<half, 2>()
+        //&& testVArithmetic<half, 3>()
+        //&& testVArithmetic<half, 4>()
```

Comment:
Let's test only four types as the spec requires,
        genType  subgroupAdd(genType value);
        genIType subgroupAdd(genIType value);
        genUType subgroupAdd(genUType value);
        genDType subgroupAdd(genDType value);
That is float, int, uint, double and each of them has vector2,3,4 variants.

Reply:
also from https://github.com/KhronosGroup/GLSL/blob/main/extensions/ext/GLSL_EXT_shader_subgroup_extended_types.txt

we will need half's eventually

if you do want these removed regardless I can do this

Reply:
Commented out lines in code can easily rot.
You can comment out certain test cases as a part of quick iterations of TDD method.
But it is not ideal to submit them.

I understand that we will need them eventually.
But we can have them when they are actually working.
We should either delete those or make them working.